app/models/application_record.rb

```diff
@@ -1,3 +1,21 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  QUERY_ESTIMATED_COUNT = <<~SQL.squish.freeze
+    SELECT (
+      (reltuples / GREATEST(relpages, 1)) *
+      (pg_relation_size($1) / (GREATEST(current_setting('block_size')::integer, 1)))
+    )::bigint AS count
+    FROM pg_class WHERE relname = $2;
+  SQL
+
+  # Computes an estimated count of the number of rows using stats collected by VACUUM
+  # inspired by <https://www.citusdata.com/blog/2016/10/12/count-performance/#dup_counts_estimated_full>
+  # and <https://stackoverflow.com/a/48391562/4186181>
+  def self.estimated_count
+    query_name = "SQL COUNT ESTIMATE: #{table_name}"
+    table_name_attr = ActiveRecord::Relation::QueryAttribute.new("relname", "users", ActiveRecord::Type::String.new)
```

Comment:
You will want to replace `"users"` here with `table_name` so it works with any model.

Reply:
Great catch!

Done:

```ruby
[1] pry(main)> Article.count
   (25.2ms)  SELECT COUNT(*) FROM "articles"
=> 26
[2] pry(main)> Article.estimated_count
  SQL COUNT ESTIMATE: articles (16.8ms)  SELECT ( (reltuples / GREATEST(relpages, 1)) * (pg_relation_size($1) / (GREATEST(current_setting('block_size')::integer, 1))) )::bigint AS count FROM pg_class WHERE relname = $2;  [["relname", "articles"], ["relname", "articles"]]
=> 25
```